### ISSUE TITLE
dropout after the attention

### ISSUE THREAD

**Issue by Howuhh** [typoverflow/UtilsRL#32]
### Prerequisites

- [X] I have read the latest version of source code.
- [X] I have checked both open and closed issues but found nothing that helps.

### UtilsRL version

-

### System info

-

### Your question

Hi! Saw a question in the code about the dropout after the attention here:
https://github.com/typoverflow/UtilsRL/blob/698414465535d6fd082c830ea82c50a6a73a896c/UtilsRL/net/attention.py#L63

Main reference for me (and for implementation in corl) was minGPT and nanoGPT implementations by @karpathy, where dropout is applied after the attention:
https://github.com/karpathy/minGPT/blob/37baab71b9abea1b76ab957409a1cc2fbfba8a26/mingpt/model.py#L70

### Urgency

Not a big problem.

**Comment by typoverflow** [comment]
Yes this is a known bug. The root cause is pytorch MHV don't add dropout after its final projection layer, although minGPT and nanoGPT do dropout in their implementations of causal attention. I have fixed this issue in https://github.com/typoverflow/OfflineRL-Lib/tree/master/offlinerllib/module/net/attention, and for UtilsRL, there is an unmerged branch named `fix_dt` which deals with this issue. The bug fix branch is unmerged because we are going to add some new Transformer architectures, such as [RWKV](https://github.com/BlinkDL/RWKV-LM). 

Anyway, thanks for helping to check the code and reaching out. There are still some bugs in this implementation, such as the missing layer norm for DT which we have discussed in https://github.com/tinkoff-ai/CORL/issues/40. I'm launching some experiments before finally merging the bug-fix branch to the master : )
